### ISSUE TITLE
print output to console by default

### ISSUE THREAD

**Issue by fpvandoorn** [leanprover/tutorial#149]
I think we should print the output to the console by default.

I spent the last hour trying to figure out how to do this, but I failed, because I don't know any javascript, and could only figure out so much in an hour.

Here is my best attempt: 85619439b0ce2611db7a2b1cfc19a0c78f270d6c. I tried to invert the meaning of the cookie, because an non-existing cookie is interpreted as `false`. But after this change, if the value of the cookie is `false` (meaning that you set the option "print output to console" to `true`), and you reload the page, the option "print output to console" is `false` again.

Could you do this Soonho (if you agree that this is a better default value)?

**Comment by soonhokong** [comment]
> Could you do this Soonho (if you agree that this is a better default value)?

No problem.

**Comment by fpvandoorn** [comment]
Thanks!

Basic javascript question: in the line
```
print_output_to_console: $.cookie("leanjs_print_output_to_console") || true, 
```
`||` is just the boolean `or`, right? So basically now you ignore the value of the cookie?

**Comment by soonhokong** [comment]
> || is just the boolean or, right?

Right.

> So basically now you ignore the value of the cookie?

No. It's using short-circuit semantics of `||`. When a user is not set the value of `leanjs_print_output_to_console`, we have `$.cookie("leanjs_print_output_to_console") = undefined` which is interpreted as `false`. Then the whole expression gives `true`.

However, when the value of the cookie `leanjs_print_output_to_consloe` is set, it returns either `"true"` or `"false"`. Note that both of them are non-null strings and interpreted as the Boolean value `true`. Because of the short-circuit semantics of `||`, it simply returns the value of the cookie.

In short, we have three cases:
 - `$.cookie("leanjs_print_output_to_console") || true` => `"true" || true` => `true`
 - `$.cookie("leanjs_print_output_to_console") || true` => `"false" || true` => `false`
 - `$.cookie("leanjs_print_output_to_console") || true` => `undefined || true` => `true`

**Comment by fpvandoorn** [comment]
Ah, that makes sense (but I would have never guessed it myself). Thanks for the explanation. If you don't mind a follow-up question: do you mean
* `"true" || true` => `"true"`
* `"false" || true` => `"false"`
* `undefined || true` => `true`

or am I still misunderstanding it?

**Comment by soonhokong** [comment]
That's what I meant (I updated my comment after I posted it). 

In short, the following is the case 
> All false, 0, empty strings '' and "", NaN, undefined, and null are always evaluated as false; everything else is true.

**Comment by soonhokong** [comment]
FYI, another interesting example which shows the oddity of JS is https://dorey.github.io/JavaScript-Equality-Table . It should be the case that only the diagonal elements are **green**. But in the world of JS, it's not the case..

**Comment by fpvandoorn** [comment]
Thanks! and LOL!

**Comment by fpvandoorn** [comment]
I "like" how `==` is not transitive and `a==b` doesn't imply that `if(a)` and `if(b)` do the same thing.

One last question: If the value of `print_output_to_console` is `"false"`, doesn't that mess up the test

`if (myModule.print_output_to_console)`

because `"false"` is coerced to `true`? This looks like what is happening: if I set the option to false and refresh the page, then the value is still false, but the output is printed to the console.

**Comment by soonhokong** [comment]
I see. I'll check it.

**Comment by soonhokong** [comment]
@fpvandoorn: when a JS value is saved into cookie, its string representation is saved. As a result, we may have two choices for the type of `myModule.print_output_to_console` -- either `string` or `boolean`. I think it's better to keep it as `boolean`. To do so, when we load a cookie value for `print_output_to_console`, we cast a loaded value (: `string`) into `boolean`. See e7658c4.
